binderhub/redirect.py

```diff
@@ -76,14 +76,33 @@ def username_from_image(self, image):
         # start server
         app_log.info("Starting server for user %s with image %s", username, image)
         try:
-            await self.api_request(
+            resp = await self.api_request(
                 'users/%s/server' % username,
                 method='POST',
                 body=json.dumps({
                     'token': token,
                     'image': image,
                 }).encode('utf8'),
             )
+            if resp.code == 202:
+                # Server hasn't actually started yet
+                # We wait for it!
+                for i in range(10):
+                    resp = await self.api_request(
+                        'users/%s' % username,
+                        method='GET',
+                    )
+
+                    body = json.loads(resp.body.decode('utf-8'))
+                    if body['server']:
+                        break
+                    # FIXME: make this configurable
+                    # FIXME: Measure how long it takes for servers to start
+                    # and tune this appropriately
+                    await gen.sleep(1.4 ** i)
```

Comment:
why 1.4? is 14 some kind of special dev-ops maximum wait time?  +1 to being able to set this in the helmchart

Reply:
I played with values between 1 and 2 and found 1.4 to provide total wait times that weren't too large but also not too small. Agree it needs to be settable in helm chart.